File: Python/Dosen.py
from SivitasAkademik import SivitasAkademik


class Dosen(SivitasAkademik):
    __nip = ""
    __pend_terakhir = ""
    __keahlian = ""

    def __init__(self, nip="", pend_terakhir="", keahlian="", nik="", nama="", jenis_kelamin="", asal_universitas="", email_edu=""):
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu)
        self.__nip = nip
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian

    def set_nip(self, nip):
        self.__nip = nip

    def set_pend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir

    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

    def get_nip(self):
        return self.__nip

    def get_pend_terakhir(self):
        return self.__pend_terakhir

    def get_keahlian(self):
        return self.__keahlian

File: C++/Dosen.cpp
#include "header.hh"
Dosen::Dosen() : SivitasAkademik()
{
    this->nip = "";
    this->pend_terakhir = "";
    this->keahlian = "";
}
Dosen::Dosen(string nip, string pend_terakhir, string keahlian, string nik, string nama, string jenis_kelamin, string asal_universitas, string email_edu) : SivitasAkademik(nik, nama, jenis_kelamin, asal_universitas, email_edu)
{
    this->nip = nip;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}
void Dosen::set_nip(string nip)
{
    this->nip = nip;
}
void Dosen::set_pend_terakhir(string pend_terakhir)
{
    this->pend_terakhir = pend_terakhir;
}
void Dosen::set_keahlian(string keahlian)
{
    this->keahlian = keahlian;
}
string Dosen::get_nip()
{
    return this->nip;
}
string Dosen::get_pend_terakhir()
{
    return this->pend_terakhir;
}
string Dosen::get_keahlian()
{
    return this->keahlian;
}
Dosen::~Dosen()
{
}